setup.py

```diff
@@ -80,7 +80,7 @@
             f"snowflake-connector-python[secure-local-storage]{CONNECTOR_DEPENDENCY_VERSION}",
         ],
         "development": [
-            "pytest",
+            "pytest<8.0.0",
```

Comment:
pytest released v8 last Friday and it's causing flakiness in the following test:

`FAILED tests/integ/test_dataframe.py::test_write_temp_table_no_breaking_change[append-temp] - AssertionError: assert 'create_temp_table is deprecated' in ''`

https://github.com/snowflakedb/snowpark-python/actions/runs/7701541832

Reply:
opened an issue in pytest: https://github.com/pytest-dev/pytest/issues/11886
created a jira SNOW-1022240 to follow up